### PR TITLE
Allow Redis Serializer option to be set outside

### PR DESCRIPTION
**Pull request by yellow1912**
The RedisCache Adapter should not force Serializer Option.
The Serializer Option should be set When the redis client
is created (outside of the scope of this class).

### CODE REVIEW

File: lib/Doctrine/Common/Cache/RedisCache.php [hunk 1]
```diff
@@ -44,7 +44,11 @@ class RedisCache extends CacheProvider
      */
     public function setRedis(Redis $redis)
     {
-        $redis->setOption(Redis::OPT_SERIALIZER, $this->getSerializerValue());
+        // The redis object is injected from an external service
+        // Serializer option should be set outside
+        // We should not force the serializer option here
+        // This code is left here for reference only
```
Comment (stof) [inline lib/Doctrine/Common/Cache/RedisCache.php]: code should not be left for reference. It should be removed if it is not used anymore (I cannot tell whether the doctrine/cache maintainers will accept it or no though)

Comment (yellow1912) [inline lib/Doctrine/Common/Cache/RedisCache.php]: Thanks stof, I will wait for the response from the maintainer then if the change is ok I will remove the reference code as you suggest.

**Comment by Tobion** [comment]
While I agree that it is not be responsibility of the class to set this option (esp. since it cannot be overwritten), changing this is probably a big BC break. Also it makes the method `getSerializerValue` unused and misplaced.

**Comment by Tobion** [comment]
Ref. #150

**Comment by yellow1912** [comment]
Isn't getSerializerValue a protected value only used for this class? Can't we safely drop it?

**Comment by mvanbaak** [comment]
@yellow1912 Can you pick this up? If not, I can create a new PR. (We are affected by this as well, and removing the forcing of the serializer makes the RedisCache in line with MemcachedCache)

**Comment by mvanbaak** [comment]
Created new PR which removes the getSerializerValue method and directly sets the passed Redis object to $this->redis (now in line with MemcachedCache)

**Comment by yellow1912** [comment]
Sorry I was busy. I see you already created new PR, cant wait to see it get merged :)

**Comment by mvanbaak** [comment]
:) I know how that is. Since we are using this fix in production for a week now I thought it was a good idea to share it publicly.

**Comment by lcobucci** [comment]
As discussed in #211 we would break BC with that change.
Closing this to use #211 as recommended solution.

**Pull request closed without merging**